sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -338,8 +339,12 @@ def database_identifier(self):
                 tag_dict['azure_name'] = self.resolved_hostname[: -len(AZURE_SERVER_SUFFIX)]
             if self.static_info_cache.get(STATIC_INFO_SERVERNAME) is not None:
                 tag_dict['server_name'] = self.static_info_cache.get(STATIC_INFO_SERVERNAME)
-            if self.static_info_cache.get(STATIC_INFO_INSTANCENAME) is not None:
-                tag_dict['instance_name'] = self.static_info_cache.get(STATIC_INFO_INSTANCENAME)
+                # in many cases the instance name is not set, so we use an empty string as a fallback in the
+                # cases where we already have a server name. This usually means that the instance is not a named instance
+                # and the server name is the same as the instance name.
+                tag_dict['instance_name'] = self.static_info_cache.get(STATIC_INFO_INSTANCENAME) or ''
```

Comment:
This is a change from how all other variables work. If we want to introduce defaults we can but let's do it holistically rather than as a one off that will increase confusion when it behaves differently than every other variable